File: src/server/logic/WinConditions.java
package server.logic;

import shared.model.Player;
import shared.model.Role;
import java.util.Collection;

public class WinConditions {
    public static String checkWin(Collection<Player> players) {
        int wolvesCount = 0;
        int villagersCount = 0;

        for (Player p : players) {
            if (p.isAlive() && p.isConnected()) {
                if (p.getRole() == Role.WEREWOLF) wolvesCount++;
                else villagersCount++;
            }
        }

        if (wolvesCount == 0) return "VILLAGERS_WIN";
        if (wolvesCount >= villagersCount) return "WEREWOLVES_WIN";
        return "NO_WINNER_YET";
    }
}

File: src/server/logic/WinConditions.cs
using System.Collections.Generic;
using WerewolfGame.Shared.Model;

namespace WerewolfGame.Server.Logic
{
    public static class WinConditions
    {
        public static string CheckWin(IEnumerable<Player> players)
        {
            int wolvesCount = 0;
            int villagersCount = 0;

            foreach (var p in players)
            {
                if (p.IsAlive && p.IsConnected)
                {
                    if (p.Role == Role.WEREWOLF) wolvesCount++;
                    else villagersCount++;
                }
            }

            if (wolvesCount == 0) return "VILLAGERS_WIN";
            if (wolvesCount >= villagersCount) return "WEREWOLVES_WIN";
            return "NO_WINNER_YET";
        }
    }
}